src/mpid/ch4/shm/posix/posix_coll_release_gather.h

```diff
@@ -229,7 +229,7 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_POSIX_mpi_reduce_release_gather(const void *s
         MPI_Aint chunk_count = (i == 0) ? chunk_size_floor : chunk_size_ceil;
 
         mpi_errno =
-            MPIDI_POSIX_mpi_release_gather_release(NULL, 0, MPI_DATATYPE_NULL, root,
+            MPIDI_POSIX_mpi_release_gather_release(NULL, chunk_count, datatype, root,
```

Comment:
Is this fixing a previous typo? It doesn't appear to belong in this commit.

Reply:
Previously these two fields are not used in MPIDI_POSIX_mpi_release_gather_release for reduction, but as the new changes in MPIDI_POSIX_mpi_release_gather_release, these two fields need to be set correctly.